storage-provider/server/src/p2p/mod.rs

```diff
@@ -193,6 +217,56 @@ where
             _ => {}
         }
     }
+
+    #[instrument(level = "trace", skip(self))]
+    fn on_request_response_event(
+        &mut self,
+        event: RequestResponseEvent<PieceInfoRequest, PieceInfoResponse>,
+    ) {
+        match event {
+            RequestResponseEvent::Message { peer, message } => {
+                if let Message::Request {
+                    request,
+                    channel,
+                    request_id,
+                } = message
+                {
+                    trace!("Got request with id {request_id} from {peer}");
+
+                    let response = match self.index_db.get_piece_metadata(request.piece_cid) {
+                        Ok(info) => PieceInfoResponse::Found(PieceInfo { roots: info.roots }),
+                        Err(err) => {
+                            error!(?err, "error occurred while retrieving piece info");
+                            PieceInfoResponse::NotFound(request.piece_cid)
+                        }
+                    };
+
+                    if self
+                        .swarm
+                        .behaviour_mut()
+                        .request_response
+                        .send_response(channel, response)
+                        .is_err()
+                    {
+                        error!("Failed to send piece info to {peer:?}");
+                    }
+                }
+            }
+            RequestResponseEvent::OutboundFailure {
+                peer,
+                request_id,
+                error,
+            } => warn!("Failed to send response with id {request_id} to {peer}: {error}"),
+            RequestResponseEvent::InboundFailure {
+                peer,
+                request_id,
+                error,
+            } => warn!("Failed to receive message with id {request_id} from {peer}: {error}"),
+            RequestResponseEvent::ResponseSent { peer, request_id } => {
+                trace!("Response with id {request_id} sent to {peer}")
```

Comment:
We can discuss this later but IMO `trace` is more "yolo-logging" where you put it everywhere, this seems more like a `debug`
```suggestion
                debug!("Response with id {request_id} sent to {peer}")
```